test_plans/barriers_fences.asciidoc

```diff
@@ -0,0 +1,74 @@
+:sectnums:
+:xrefstyle: short
+
+= Test plan for barriers and fences

```

Comment:
We already have a test plan for barriers: https://github.com/KhronosGroup/SYCL-CTS/blob/SYCL-2020/test_plans/group_functions.asciidoc#252-group_barrier

If that part is duplicated, then this plan could cover just `atomic_fence`.